dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/file/DeferredWriterTest.kt

```diff
@@ -77,52 +83,143 @@ internal class DeferredWriterTest {
         inOrder.verify(mockDelegate).write(model)
     }
 
+    @Test
+    fun `migrates the data before doing anything else in multi thread`(forge: Forge) {
+        val model1 = forge.anAlphabeticalString()
+        val model2 = forge.anAlphabeticalString()
+        val model3 = forge.anAlphabeticalString()
+
+        val countDownLatch = CountDownLatch(3)
+
+        // when
+        Thread {
+            underTest.write(model1)
+            countDownLatch.countDown()
+        }.start()
+        Thread {
+            underTest.write(model2)
+            countDownLatch.countDown()
+        }.start()
+        Thread {
+            underTest.write(model3)
+            countDownLatch.countDown()
+        }.start()
+
+        countDownLatch.await(3000, TimeUnit.SECONDS)
+
+        // then
+        val inOrder = inOrder(mockDataMigrator, mockDelegate)
```

Comment:
`inOrder` is only useful if you assert several calls in order. Here only  the mockDelegate.write is checked in order. 
You can write : 
```kotlin
inOrder(mockDataMigrator, mockDelegate) {
    verify(mockDataMigrator).migrateData()
    argumentCaptor<String> {
        verify(mockDelegate, times(3)).write(argCaptor.capture())
        assertThat(allValues).containsExactlyInAnyOrder(model1, model2, model3)
    }
}
```

Reply:
This doesn't really change much the test, you can change it later

Reply:
I will remove that....I actually copy paste those from above and forgot to adapt/change them